tests/integration/questionnaire/test_questionnaire_previous_link.py

```diff
@@ -0,0 +1,75 @@
+from tests.integration.create_token import create_token
+from tests.integration.integration_test_case import IntegrationTestCase
+
+
+class TestQuestionnairePreviousLink(IntegrationTestCase):
+
+    def test_previous_link_doesnt_appear_on_introduction(self):
+        # Given
+        token = create_token('final_confirmation', 'test')
+        resp = self.client.get('/session?token=' + token.decode(), follow_redirects=False)
+
+        # When we open the introduction
+        content = resp.get_data(True)
+
+        # Then previous link does not appear
+        self.assertNotRegexpMatches(content, 'Previous')
+
+    def test_previous_link_appears_on_all_non_start_end_pages(self):
```

Comment:
What is a non-start end page? I wonder if there is any benefit of splitting this out into two tests for clarity? e.g. `test_previous_link_not_visible_on_page_immediately_after_introduction` and `test_previous_link_visible_on_subsequent_question_pages` or similar.